lcls-twincat-pmps/Library/MajorComponents/Arbiter/FB_Arbiter.TcPOU

```diff
@@ -110,29 +121,91 @@ ArbitrateBP.nRate := MIN(stBP1.nRate, stBP2.nRate);]]></ST>
       <Declaration><![CDATA[(* Searches state assertion pool for a state-id, and sets q_xStateIDFound true if found.
 
 Use like so:
-
-fbArbiter.A_VerifyAssertion(
-i_nStateId := nStateIDAssertionToCheck
-);
-
-IF fbArbiter.q_xStateIDFound THEN:
-    Do something
+IF fbArbiter.CheckRequest(nStateIDAssertionToCheck) THEN:
+    Request is found and active in arbitration,. Do something.
 ELSE:
-    Don't do something
+    Request was not found, or is not yet included in arbitration. Don't do something/ wait.
 
 *)
 METHOD CheckRequest : BOOL
 VAR_INPUT
     nReqID    :    DWORD;
 END_VAR
+VAR
+    BP : ST_BeamParams;
+END_VAR]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[fbBPAssertionPool.A_Lookup(key := nReqID);
+
+// Verify BP is acknowledged
+BP := fbBPAssertionPool.getValue;
+
+// This logic:
+// Did we find the assertion in the pool?
+// Is the assertion cohort number less than the current cohort and greater than zero?
+// Is the Aribter itself active in arbitration?
+CheckRequest := fbBPAssertionPool.bOk AND 
+                BP.nCohortInt < nArbiterCohort AND
+                BP.nCohortInt > 0 AND 
+                xActiveInArbitration;
+]]></ST>
+      </Implementation>
+    </Method>
+    <Method Name="CheckRequestInPool" Id="{f0776990-a211-400b-9974-8651dfd1d437}">
+      <Declaration><![CDATA[// Verify request is at least in the pool
+// Does not verify request has been included in arbitration.
+// Use CheckRequest instead.
+METHOD CheckRequestInPool : BOOL
+VAR_INPUT
+    nReqID    :    DWORD;
+END_VAR
 ]]></Declaration>
       <Implementation>
         <ST><![CDATA[THIS^.fbBPAssertionPool.A_Lookup(key := nReqID);
 
-CheckRequest := THIS^.fbBPAssertionPool.bOk;
+CheckRequestInPool := THIS^.fbBPAssertionPool.bOk;]]></ST>
+      </Implementation>
+    </Method>
+    <Method Name="ElevateRequest" Id="{7ff662cb-d23f-49f5-854d-1d5b340251d9}">
+      <Declaration><![CDATA[// Elevates the arbitrated BP set to something above.
+// Could be another arbiter, or a BP requester/ IO, 
+// or an FB that locks in a specific portion of the BP set.
+METHOD ElevateRequest : BOOL
+VAR_INPUT
+    HigherAuthority : ITF_HigherAuthority;
```

Comment:
Additional comment bordering on pedantry - I think the interface prefix is usually `I_`:
Twincat example: `INTERFACE | I_ | Interface | I_CylinderControl`